src/nodes/chunk_dispatch_state.c

```diff
@@ -142,14 +143,50 @@ chunk_dispatch_exec(CustomScanState *node)
 	 * the es_result_relation_info this has to be updated every time, not
 	 * just when the chunk changes.
 	 */
-	estate->es_result_relation_info = cis->result_relation_info;
+	if (cis->compress_state != NULL)
+		estate->es_result_relation_info = cis->orig_result_relation_info;
+	else
+		estate->es_result_relation_info = cis->result_relation_info;
 
 	MemoryContextSwitchTo(old);
 
 	/* Convert the tuple to the chunk's rowtype, if necessary */
 	if (cis->hyper_to_chunk_map != NULL)
 		slot = execute_attr_map_slot(cis->hyper_to_chunk_map->attrMap, slot, cis->slot);
 
+	if (cis->compress_state != NULL)
+	{
+		/*
+		 * During the insert BEFORE ROW triggers defined on the compressed
+		 * chunk will get executed as part of postgres INSERT processing.
+		 * To support BEFORE ROW insert trigger defined on the uncompressed chunk
+		 * we have to explicitly execute those triggers.
```

Comment:
compressed chunk refers to the internal private chunk here. The user defines triggers on the hypertable and those are inherited by the chunks. Since we redirect inserts/copies to the internal compressed chunk, we need special logic to execute the triggers  ( constraints etc). on the "base" chunk. This code handles that.

Reply:
modified the comment. Hopefully that makes it clearer.